lib/archethic/account/mem_tables/token_ledger.ex

```diff
@@ -194,4 +245,56 @@ defmodule Archethic.Account.MemTables.TokenLedger do
         end)
     end
   end
+
+  @doc """
+  Spend all the unspent outputs for the given address
+  """
+  @spec spend_all_unspent_outputs(address :: binary(), from_init? :: boolean()) :: :ok
+  def spend_all_unspent_outputs(address, from_init? \\ false) do
+    case :ets.lookup(@unspent_output_index_table, address) do
+      [] ->
+        :ok
+
+      [{^address, true}] ->
+        :ets.delete(@unspent_output_index_table, address)
+        :ok
+
+      utxos ->
+        write_to_db(address, utxos, from_init?)
+    end
+  end
+
+  defp write_to_db(address, inputs, false) do
+    {:ok, pid} = DB.start_inputs_writer(:token, address)
+
+    Enum.each(inputs, fn {to, from, token_address, token_id} ->
+      [{_, amount, _spent, timestamp, protocol_version}] =
+        :ets.lookup(@ledger_table, {to, from, token_address, token_id})
+
+      DB.append_input(pid, %VersionedTransactionInput{
+        protocol_version: protocol_version,
+        input: %TransactionInput{
+          from: from,
+          amount: amount,
+          spent?: true,
```

Comment:
This is not always true, if you reach the threshold, you didn't spend the input, you just write it to disk

Reply:
since spent flag is not use anymore, as we are using now `Archethic.get_inputs`  and determine spent.
- as discussed we need another issue for it
- another issue to remove get_unpsent_outputs 
- to update the get_balance use Archethic.get_inputs.

Reply:
It just that writting doesn't mean always spent. Particularly with the threshold memory limitation. 

And I don't think we should remove the get_unspent_outputs function.